### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by elmar8287**
- Add drag and drop functionality
- Add delete button for each task

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ let tasks; // this is an empty array of our taks
 //   index: 1
 // }
```
Comment (mimipeshy) [inline src/index.js]: - [x] Kindly remove all commented out code to keep your code base clean

File: src/index.js [hunk 2]
```diff
@@ -44,12 +52,82 @@ const updateLocal = () => { // add to Local storage
...
+  }, 500);
+};
+
+// const editTask = index => {
+//   let span = tasks[index].description;
+//   // todoItemElems[index].classList.add('edition');
+//   let aaa = document.createElement('input');
+//   aaa.type = 'text';  
+//   span = aaa.value;
+// };
```
Comment (mimipeshy) [inline src/index.js]: - [x] Kindly remove commented out code to keep your codebase clean

File: src/index.js [hunk 3]
```diff
@@ -44,12 +52,82 @@ const updateLocal = () => { // add to Local storage
...
+      return {
+        offset: offset,
+        element: child
+      }
+    } else {
+      return closest
+    }
+  }, {
+    offset: Number.NEGATIVE_INFINITY
+  }).element
```
Comment (mimipeshy) [inline src/index.js]: - [ ] From the project specifications listed [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md), you were supposed to create two different files  ⬇️ 
image
Kindly fix this
- [ ] When i drag an item to another index, it should update the items index as stated here ⬇️ 
image
```
Example
I have three todo's with different indexes, A,B,C. If i move C to replace B, C's new index should be B's previous index. 
```
Hope that is clear enough. Kindly work on it as well

- [x] When i click on completed on a task, it should change from `false` to `true` and vice versa. Kindly work on this too

- [ ] From the Learning Objectives listed [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md) the last point states that you should use ES6 modules to write modular js. This means using `arrow functions` over the `function` keyword. Kindly fix this also

**Review by mimipeshy — CHANGES_REQUESTED** [review]
## Hi @elmar8287  👋🏾 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
#### Highlights

- items render dynamically on page load ✔️ 
- i can drag items easily ✔️ 
- good GitHub workflow ✔️ 
- not forming any linter errors ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

**Review by de-scholar — APPROVED** [review]
Hi @elmar8287 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- [ ] I would still suggest you remove [these commented codes](https://github.com/elmar8287/todo-list/blob/a3fada0cb9382dc0988be853f0af03fde55939b4/src/index.js#L164-L166)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ZawadiSumba66 — CHANGES_REQUESTED** [review]
#### Hi @elmar8287, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 

You have submitted the interactive list project for the add_remove task project. Could it be because you submitted the project with the wrong link on your dashboard? Please rectify this and request another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by elmar8287** [comment]
Dear @de-scholar! 🥇 

Many thanks for the review. I appreciate your comments 👍 

Many thanks and regards,

Elmar

**Comment by elmar8287** [comment]
Dear @ZawadiSumba66! 🥇 

Sorry for the missunderstanding. 

Many thanks and regards,

Elmar

**Pull request merged**